mfhdf/dumper/hdp.c

```diff
@@ -87,58 +110,56 @@ init_dump_opts(dump_info_t *dump_opts)
 int
 main(int argc, char *argv[])
 {
-    command_t  cmd;       /* command to perform */
-    intn       curr_arg;  /* current cmd line argument */
-    dump_opt_t glob_opts; /* global options for all commands */
-    intn       j;         /* local counting variables */
+    command_value_t cmd = BAD_COMMAND; /* command to perform */
+    intn            curr_arg;          /* current cmd line argument */
+    dump_opt_t      glob_opts;         /* global options for all commands */
+    intn            j;                 /* local counting variables */
 
     memset(&glob_opts, 0, sizeof(dump_opt_t));
 
     if (argc < 2) {
         usage(argc, argv);
-        exit(1);
-    } /* end if */
+        exit(EXIT_FAILURE);
+    }
 
     curr_arg = 1;
-    /*  printf("Argument 0: %s\n",argv[0]);
-        printf("Argument 1: %s\n",argv[1]);
-        */
     while (curr_arg < argc && (argv[curr_arg][0] == '-')) {
-        /*  while(curr_arg<argc && (argv[curr_arg][0]=='-' || argv[curr_arg][0]=='/')) {  */
         switch (argv[curr_arg][1]) {
             case 'H':
-                /*     case 'h':  */ /*    Print help for a given command */
+                /* case 'h':  */ /*    Print help for a given command */
                 if (curr_arg < argc - 1) {
                     glob_opts.help = TRUE; /* for displaying options. */
                     break;
                 }
             default:
                 usage(argc, argv); /* Display the general usage. */
-                exit(1);
-        } /* end switch */
+                exit(EXIT_FAILURE);
+        }
         curr_arg++;
-    } /* end while */
+    }
 
-    for (j = 0, cmd = HELP; j < (sizeof(commands) / sizeof(const char *)); j++, cmd++) {
-        if (HDstrcmp(argv[curr_arg], commands[j]) == 0)
+    for (j = 0; j < (sizeof(commands) / sizeof(command_t)); j++) {
+        if (HDstrcmp(argv[curr_arg], commands[j].name) == 0) {
+            cmd = commands[j].value;
             break;
-    } /* end for */
+        }
+    }
+    if (cmd == BAD_COMMAND) {
+        printf("Invalid command: %s\n", argv[curr_arg]);
+        exit(EXIT_FAILURE);
+    }
```

Comment:
Before we printed the value of cmd (in code below), which would just be the sentinel NONE value, which is unhelpful.